### ISSUE TITLE
Newest versions of numpy (>=1.20.0) do not work

### ISSUE THREAD

**Issue by clark2668** [nu-radio/NuRadioMC#533]
**Describe the issue**

Numpy deprecated a bunch of aliases as of 1.20.0. See [here](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations) for more info. This means that any `np.float` distributed throughout the code (e.g. in the [generator](https://github.com/nu-radio/NuRadioMC/blob/develop/NuRadioMC/EvtGen/generator.py#L1233)) throw errors like the one below:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File ".../miniconda/lib/python3.10/site-packages/numpy/__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'cfloat'?
```

**Expected behavior**
The import should succeed.

**Additional context**
None to add.

**Comment by clark2668** [comment]
As an aside, I'm not sure why this wasn't caught by the build-bot. Is the build bot somehow accidentally invoking old python or old numpy?

**Comment by sjoerd-bouma** [comment]
Thanks Brian for flagging this. The CI runs on Python 3.7, where pip resolves the dependencies to install numpy 1.21.6, it seems, and np.float has only started raising an error since numpy 1.24. Should of course be fixed anyway.

**Comment by clark2668** [comment]
No problem. Thanks for the really rapid response Sjoerd!

**Comment by sjoerd-bouma** [comment]
#538 fixed this in develop, but #534 (into `master`) didn't include `np.complex` and `np.bool` yet. We may want to merge #538 into master as well (will have a look next week).

**Comment by sjoerd-bouma** [comment]
This is fixed as of v2.2.0
